code/firmware/rosco_m68k_firmware/stage2/load/load.c

```diff
@@ -15,17 +15,39 @@
 
 #include <stdbool.h>
 #include <stdint.h>
+#include <string.h>
+#include "elf.h"
 #include "fat_filelib.h"
 #include "part.h"
 #include "system.h"
 
+/*
+ * Currently, software that relocates itself to 0x2000 must be loaded like a
+ * raw binary program (to 0x40000), to avoid overwriting firmware stage2.
+ * If ELF_LOAD_VIRTUAL_ADDR is 1, instead load every segment to its run virtual
+ * address, and use E_ENTRY.
+ */ 
+#ifndef ELF_LOAD_VIRTUAL_ADDR
+#define ELF_LOAD_VIRTUAL_ADDR 0
+#endif
+
 extern void mcPrint(const char *str);
 extern void print_unsigned(uint32_t num, uint8_t base);
 
+typedef void (*KMain)(volatile SystemDataBlock * const);
```

Comment:
Is this now in two places (this file, and the one below)? If so, just wondering if it's worth moving it to a header or something?